chart/kubeapps/templates/apprepository/apprepositories-secret.yaml

```
@@ -7,7 +7,7 @@ metadata:
   {{- if .namespace }}
   namespace: {{ .namespace | quote }}
   {{- else }}
-  namespace: {{ $.Release.Namespace | quote }}
+  namespace: {{ printf "%s-%s" $.Release.Namespace $.Values.apprepository.globalReposNamespaceSuffix | quote }}
```

Regarding making this a non-breaking change, could we instead do:

```suggestion
  namespace: {{ printf "%s%s" $.Release.Namespace $.Values.apprepository.globalReposNamespaceSuffix | quote }}
```
so that if `globalReposNamespaceSuffix` is empty (which can be the default, initially), we will have the existing behavior? We can then update the default value to `-repos-global` when we do a major version change, breaking the backwards compatibility?

It'd require a conditional around the namespace creation later too, etc.

---

Thanks for the suggestion Michael, it makes sense. So, defaulting now to the same `kubeapps` namespace, would mean leaving without effect the purpose of this change until next major version 3.x?
As-is in a Kubeapps upgrade scenario, new global repos will go to the new ns and old global repos will stay in the `kubeapps` ns, just not being considered global anymore.

---

> Thanks for the suggestion Michael, it makes sense. So, defaulting now to the same `kubeapps` namespace, would mean leaving without effect the purpose of this change until next major version 3.x?

By default, yes, but we can verify it works as expected, even update CI to use it (by setting a value for the suffix) if we want, to ensure that switching the default later is a non-event (other than the instructions for users).

> As-is in a Kubeapps upgrade scenario, new global repos will go to the new ns and old global repos will stay in the `kubeapps` ns, just not being considered global anymore.

Yep, we'll just ensure we document that in the release notes so people can move their repos as needed.